opcua/common/xmlexporter.py

```diff
@@ -404,8 +407,20 @@ def _extobj_to_etree(self, val_el, name, dtype, val):
         id_el.text = dtype.to_string()
         body_el = Et.SubElement(obj_el, "uax:Body")
         struct_el = Et.SubElement(body_el, "uax:" + name)
-        for name, vtype in val.ua_types.items():
-            self.member_to_etree(struct_el, name, ua.NodeId(getattr(ua.ObjectIds, vtype)), getattr(val, name))
+        items_keys = val.ua_types.keys()
+        skip_empty = False
+        if dtype == ua.NodeId(ua.ObjectIds.Argument):
+            skip_empty = True
+            items_keys = [name for name in ['Name',
+                                            'DataType',
+                                            'ValueRank',
+                                            'ArrayDimensions',
+                                            'Description'] if name in items_keys ]
```

Comment:
That isn't so strange: Just like `Argument.to_binary` also requires an fixed order of the members, the xmls schema (https://opcfoundation.org/UA/2008/02/Types.xsd) also requires an fixed order.

In a lot of cases the order doesn't matter, but if a tool checks the import against the scheme the xml parse will fail.